ports/nrf/common-hal/busio/SPI.c

```diff
@@ -224,21 +232,36 @@ void common_hal_busio_spi_unlock(busio_spi_obj_t *self) {
 }
 
 bool common_hal_busio_spi_write(busio_spi_obj_t *self, const uint8_t *data, size_t len) {
-    if (len == 0)
+    if (len == 0) {
         return true;
+    }
+
+    const bool is_spim3 = self->spim_peripheral->spim.p_reg == NRF_SPIM3;
 
     const uint32_t max_xfer_size = self->spim_peripheral->max_xfer_size;
     const uint32_t parts = len / max_xfer_size;
     const uint32_t remainder = len % max_xfer_size;
 
     for (uint32_t i = 0; i < parts; ++i) {
-        const nrfx_spim_xfer_desc_t xfer = NRFX_SPIM_XFER_TX(data + i * max_xfer_size, max_xfer_size);
+        uint8_t *start = (uint8_t *) (data + i * max_xfer_size);
+        if (is_spim3) {
+            // If SPIM3, copy into unused RAM block, and do DMA from there.
+            memcpy(spim3_transmit_buffer, start, max_xfer_size);
```

Comment:
I wonder if we should put the displayio buffer in this ram as well. That way we could check the location of the buffer and skip the memcpy.

Reply:
That sounds like a good idea, since the 8kB is otherwise wasted. In fact, in the current implementation, we are only using the first 8 bytes or something like that. But we have to allocate the whole 8kB because otherwise there could be errors caused by contention on the block.

The buffer would be in that space conditionally; you'd check that the SPIM is SPIM3 when making the decision.